lib/snappy_ex.ex

```diff
@@ -0,0 +1,123 @@
+defmodule SnappyEx do
+  import Bitwise
+
+  @moduledoc """
+    SSZ library in Elixir
+  """
+  @bit_mask_32 2 ** 31 - 1
+  @chunk_size_limit 65_540
+
+  def decompress_frames(chunks) do
+    with {:ok, chunks} <- validate_stream(chunks) do
+      decompress_frames(chunks, <<>>)
+    end
+  end
+
+  defp decompress_frames(chunks, acc) when byte_size(chunks) == 0, do: acc
+
+  defp decompress_frames(chunks, acc) do
+    with {:ok, {chunk_id, chunk_size, chunks}} <- process_chunk_metadata(chunks),
+         {:ok, {acc, chunks}} <- process_chunks(chunks, acc, chunk_size, chunk_id) do
+      decompress_frames(chunks, acc)
```

Comment:
This looks great, but I suggest a slightly different division:

```elixir
with {:ok, {chunk_id, data, remaining_chunks}} <- process_chunk_metadata(chunks),
     {:ok, new_acc} <- apply_chunk(acc, id, data) do
  decompress_frames(remaining_chunks, new_acc)
end 
```

The main difference is that the size check and the chunk data extraction is common to all types, so it should be part of process_chunk_metadata. Also, apply_chunk doesn't need the size or the remaining chunks.